packages/components/MenuButton/macos/MenuButton.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+@objc(MSFMenuButton)
+open class MenuButton: NSPopUpButton {
+
+  public override init(frame buttonFrame: NSRect, pullsDown flag: Bool) {
+
+    super.init(frame: buttonFrame, pullsDown: flag)
+
+    imagePosition = .imageLeading
+    bezelStyle = .recessed
+
+    if #available(OSX 11.0, *) {
+      controlSize = .large
+    }
+
+    guard let dropDownCell = cell as? NSPopUpButtonCell else {
+      preconditionFailure()
+    }
+    dropDownCell.imagePosition = .imageLeading
+    dropDownCell.arrowPosition = .arrowAtBottom
+  }
+
+  @available(*, unavailable)
+  required public init?(coder decoder: NSCoder) {
+    preconditionFailure()
+  }
+
+  @objc public convenience init() {
+    self.init(frame: .zero, pullsDown: true)
+  }
+
+  @objc public var OnItemClick: RCTBubblingEventBlock?
+
+  @objc public var OnSubmenuItemClick: RCTBubblingEventBlock?
+
+  open override var menu: NSMenu? {
+    didSet {
+      updateMenu()
+    }
+  }
+
+  open override var image: NSImage? {
+    get {
+      return menuButtonImage
+    }
+    set {
+      // We must set the image on the dropdown cell rather than the button,
+      // If we set the image on Button itself, no image is displayed.
+      menuButtonImage = newValue
+      updateDropDownCell()
+    }
+  }
+
+  open override var title: String {
+    didSet {
+      updateDropDownCell()
+    }
+  }
+
+  // MARK: - Private Methods
+
+  private func updateMenu() {
+    guard let menu = menu else {
+      return
+    }
+
+    for (index, menuItem) in menu.items.enumerated() {
+      menuItem.target = self
+      menuItem.action = #selector(sendOnItemClickEvent)
+      if menuItem.hasSubmenu, let submenu = menuItem.submenu {
```

Comment:
The normal swift idiom would be

`if let submenu = menuItem.submenu {`

where `menuItem.submenu` would be nil if `menuItem` doesn't have a submenu.  Then we needn't bother with a `hasSubmenu` boolean.  Can we do that?

Reply:
Yeah I guess I didn't need to check `hasSubmenu`. Thanks!